solidity/dashboard/.env

```diff
@@ -0,0 +1 @@
+REACT_APP_VERSION=$npm_package_version
```

Comment:
Yes, we can, but it seems more correct to me, but I don't know why 😆. Worth mentioning that in that case(`import { version} from "package.json"`) exposes the whole of `package.json` in the build bundle, not just the version.

Reply:
> it seems more correct to me

I get where you're coming from here, actually. At first blush, pulling in the package definition to a source file within the package feels a little weird. That said, in general I consider build complexity to be a much bigger source of headaches than source complexity, because build complexity is rarely touched but also (or therefore) tends to be extremely fragile.

Playing it out for this particular use case, there's really only one place to get the right version, and that's the `package.json` file. When we do a build, there are two possibilities: we do it manually and have to set the environment variable manually, or we do it via script and have to set the environment variable via script. When we set it via script, we 
have to look it up from somewhere---and it will be the `package.json` file. So now to properly automate the build, we've still got our dependency on `package.json` and we'll probably also need `jq` to parse it properly and pull out the version, and if you do a development build you most likely won't be testing that flow unless we make the dev script also pull in the `package.json`. Now we're either maintaining two scripts that are separately managing that version variable that's surfaced in the UI, or we're only maintaining one but in develop mode we can't see if it's working or not.

I've gone a bit long, but all of this to say that for me it is indeed a little _cleaner_ to fetch it from the environment---but it ends up adding all sorts of additional complexities and problems down the line. Because we _can_ use `package.json` directly therefore, I'd argue that we _should_.

> exposes the whole of package.json in the build bundle

That's surprising but not problematic, IMO. Surprising because I would expect the app to filter that out at build time just for space efficiency (tree shaking, etc), but not problematic because the package JSON is public info anyway, so if it's all bundled in… Oh well 🤷‍♂️

Reply:
Just to be clear, the `create-react-app` take care of `$npm_package_version`. Pls [see](https://create-react-app.dev/docs/adding-custom-environment-variables#expanding-environment-variables-in-env). Note: changes to the .env file will be picked only after rebuild app.

Reply:
> the create-react-app takes care of $npm_package_version

Ah! Okay, some build magic I wasn't aware of. This… Kind of goes to my point about build magic, frankly (it's often not straightforward to answer “where is this coming from?” when something happens at build time), but it also reduces the need to worry about it right now significantly, since it's already handled externally.